src/search.rs

```diff
@@ -45,14 +45,14 @@ impl SearchTerm {
 }
 
 fn search_impl(
-    value: &JsonTreeValue,
+    value: &dyn ToJsonTreeValue,
     search_term: &SearchTerm,
     path_segments: &mut Vec<String>,
     matching_paths: &mut HashSet<Vec<String>>,
 ) {
-    match value {
+    match value.to_json_tree_value() {
```

Comment:
Could we leverage that we are converting the entire JSON when searching, to not have to do it again when rendering?